slicerator.py

```diff
@@ -91,9 +140,12 @@ def __len__(self):
     def __getattr__(self, key):
         # Remember this only gets called if __getattribute__ raises an
         # AttributeError. Try the ancestor object.
+        if key not in self._attrs:
+            raise AttributeError
         attr = getattr(self._ancestor, key)
         if isinstance(attr, Slicerator):
-            return Slicerator(attr, self.indices, len(self))
+            return Slicerator(attr, '__getitem__', self.indices, len(self),
+                              self._attrs)
```

Comment:
This is for propagating `Slicerator` attributes right? Shouldn't it be `attr._attrs`?